projects/admin/src/app/service/items.service.ts

```diff
@@ -156,7 +156,8 @@ export class ItemsService {
       map(itemData => {
         const newItem = new Item(itemData.metadata);
         newItem.actionDone = action;
-        newItem.setLoan(Object.values(itemData.action_applied).pop());
+        // Set the action applied corresponding to current action as the loan of the item
+        newItem.setLoan(itemData.action_applied[action]);
```

Comment:
Are you tested `action` doens't exists into action_applied ?
Maybe safer to create an `if (item.action_applied.hasOwnProperty(action)) { ... }` ?
But I can't reproduce this case with my scenarii :'(